test/vm-download

```diff
@@ -18,78 +18,45 @@
 # You should have received a copy of the GNU Lesser General Public License
 # along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
 
-SELF=vm-download
-. ./testlib.sh
-
-URL=http://files.cockpit-project.org/testdata/images
-
-TEST_DATA=${TEST_DATA:-$PWD}
-
-set -euf
-
-usage()
-{
-    echo >&2 "usage: $SELF [-f FLAVOR]"
-}
-
-flavor=cockpit
-
-case ${1:-} in
--f|--flavor)
-	if [ $# -lt 2 ]; then
-		usage
-		exit 2
-	fi
-	flavor=$2
-	shift 2
-	;;
---flavor=*)
-	flavor=${1#--machine=}
-	shift
-	;;
---help|-h)
-	usage
-	exit 0
-	;;
-esac
-
-os=$(python -c "print eval(open('guest/$flavor.conf', 'r').read())['os']" 2>/dev/null || true)
-os=${os:-$TEST_OS}
-
-tag=$(python -c "print eval(open('guest/$flavor.conf', 'r').read())['tags']['$os']" 2>/dev/null || true)
-tag=${tag:-0}
-
-TEST_IMAGE=$flavor-$os-$TEST_ARCH-$tag
-
-checksum_file=$TEST_IMAGE-checksum
-
-if [ ! -e "$TEST_DATA/images" ]; then
-    mkdir "$TEST_DATA/images"
-fi
-
-cd "$TEST_DATA/images"
-
-if ! curl -f -s "$URL/$checksum_file" -o "$checksum_file.remote"; then
-    if [ ! -f "$checksum_file" ]; then
-        echo Downloading $URL/$checksum_file failed.
-        exit 1
-    else
-        echo Downloading $URL/$checksum_file failed, using existing image.
-        exit 0
-    fi
-fi
-
-if [ -f "$checksum_file" ] && cmp "$checksum_file.remote" "$checksum_file"; then
-    rm -f "$checksum_file.remote"
-    exit 0
-fi
-
-cat "$checksum_file.remote" | while read sum file; do
```

Comment:
The checksum file (which is really a manifest) can contain all kinds of files in addition to the QEMU image.  For Atomic we also have a ISO for cloud-init, for example.